modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts

```diff
@@ -0,0 +1,120 @@
+import { ECDSA, Ecdsa } from './../../../account-lib/mpc/tss';
+import { DecryptableNShare, CombinedKey, SigningMaterial, EncryptedNShare } from './types';
+import { encryptAndSignText, readSignedMessage } from './../../utils';
+
+type NShare = ECDSA.NShare;
+type KeyShare = ECDSA.KeyShare;
+/**
+ * Combines NShares to combine the final TSS key
+ * This can only be used to create the User or Backup key since it requires the common keychain from BitGo first
+ *
+ * @param params.keyShare - TSS key share
+ * @param params.encryptedNShares - encrypted NShares with information on how to decrypt
+ * @param params.commonKeychain - expected common keychain of the combined key
+ * @returns {CombinedKey} combined TSS key
+ */
+export async function createCombinedKey(params: {
+  keyShare: KeyShare;
+  encryptedNShares: DecryptableNShare[];
+  commonKeychain: string;
+}): Promise<CombinedKey> {
+  const MPC = new Ecdsa();
+
+  const { keyShare, encryptedNShares, commonKeychain } = params;
+  const nShares: NShare[] = [];
+
+  let bitgoNShare: NShare | undefined;
+  let userNShare: NShare | undefined;
+  let backupNShare: NShare | undefined;
+
+  for (const encryptedNShare of encryptedNShares) {
+    const privateShare = await readSignedMessage(
+      encryptedNShare.nShare.encryptedPrivateShare,
+      encryptedNShare.senderPublicArmor,
+      encryptedNShare.recipientPrivateArmor
+    );
+
+    const nShare: NShare = {
+      i: encryptedNShare.nShare.i,
+      j: encryptedNShare.nShare.j,
+      y: encryptedNShare.nShare.publicShare,
+      u: privateShare,
+      n: encryptedNShare.nShare.n,
+      chaincode: encryptedNShare.nShare.chaincode,
+    };
+
+    switch (encryptedNShare.nShare.j) {
+      case 1:
+        userNShare = nShare;
+        break;
+      case 2:
+        backupNShare = nShare;
+        break;
+      case 3:
+        bitgoNShare = nShare;
+        break;
+      default:
+        throw new Error('Invalid NShare index');
+    }
+
+    nShares.push(nShare);
+  }
+
+  const combinedKey = MPC.keyCombine(keyShare.pShare, nShares);
+  if (combinedKey.xShare.y + combinedKey.xShare.chaincode !== commonKeychain) {
+    throw new Error('Common keychains do not match');
+  }
+  if (!bitgoNShare) {
```

Comment:
I think instead of this check, before the combined key operation takes place, we can make sure that all the nShares are there (eg: for user, both bitgo and backup nShare and for backup, both bitgo and user nshare)